src/main.py

```diff
@@ -3,20 +3,52 @@
 import httpx # Yes, we could use requests but httpx supports async tasks and HTTP/2!
 from os import getenv
 from asyncio import run
+import platform    # For getting the operating system name
+import subprocess  # For executing a shell command
 
 def get_users() -> list[str]:
+    #print(getenv('QDBotIDs'))
     return getenv('QDBotIDs').split(';')
 
+def check_ok(url: str) -> bool:
+    r = httpx.get(url)
+    #there are various status codes, duckduckgo returns 301; is there a keyword only for errors?
```

Comment:
Normally a 200 is a fulfilled request, I think that we could check if the error code is greater than 299

Reply:
Since redirects (3xx) don't mean a service is offline, I set it to <400